src/main/java/org/apache/commons/dbcp2/BasicDataSource.java

```diff
@@ -433,6 +433,55 @@ public synchronized void close() throws SQLException {
         }
     }
 
+    /**
+     * Starts the datasource.
+     * <p>
+     * It is not necessary to call this method before using a newly created BasicDataSource instance, but
+     * calling it in that context causes the datasource to be immediately initialized (instead of waiting for
+     * the first {@link #getConnection()} request). Its primary use is to restart and reinitialize a
+     * datasource that has been closed.
+     * <p>
+     * When this method is called after {@link #close()}, connections checked out by clients
+     * before the datasource was stopped do not count in {@link #getMaxTotal()} or {@link #getNumActive()}.
+     * For example, if there are 3 connections checked out by clients when {@link #close()} is invoked and they are
+     * not returned before {@link #start()} is invoked, after this method is called, {@link #getNumActive()} will
+     * return 0.  These connections will be physically closed when they are returned, but they will not count against
+     * the maximum allowed in the newly started datasource.
+     *
+     * @throws SQLException if an error occurs initializing the datasource
+     */
+    @Override
+    public synchronized void start() throws SQLException {
```

Comment:
Since `createDataSource()` returns a `DataSource`, it would seem proper for this method to return it as well.

Reply:
I don't think want to expose the internal datasource directly.  The call to createDatasource is just to get initialization to happen.  It's better for clients to use BDS itself, I think.

Reply:
Ok sounds good.